crates/apps_lib/src/client/rpc.rs

```diff
@@ -2143,16 +2143,28 @@ pub async fn query_conversions(
         .wallet()
         .await
         .get_addresses_with_vp_type(AddressVpType::Token);
-    let conversions = rpc::query_conversions(context.client())
+
+    // Download conversions from all epochs to facilitate decoding asset types
+    let mut conversions = BTreeMap::new();
+    let from = MaspEpoch::zero();
+    let to = rpc::query_masp_epoch(context.client())
         .await
-        .expect("Conversions should be defined");
+        .expect("Unable to query current MASP epoch");
+    for epoch in MaspEpoch::iter_bounds_inclusive(from, to) {
+        conversions.append(
+            &mut rpc::query_conversions(context.client(), &epoch)
+                .await
+                .expect("Conversions should be defined"),
+        );
+    }
```

Comment:
there's no reason why we cannot run these queries concurrently (e.g. with [this](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffer_unordered)). the `Stream` can be collected into a `BTreeMap` (see [`StreamExt::collect`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.collect))

Reply:
I've attempted this in 46ea724dccb9c87597bbfc50836064fb0dba7c1d even though I had to `fold` the maps instead of `collect` because of the limitations of the `Extend` trait on maps. Should we also add a couple of args for the query command to allow the user to specify the window boundary so that we can avoid querying from epoch 0 all the times?